### ISSUE TITLE
reasons for queueing proposals should be exposed in the UI

### ISSUE THREAD

**Issue by aspiers** [crowbar/crowbar#1182]
A proposal can be queued rather than immediately applied for a number of reasons, e.g. one of the required nodes is not ready, or one of the dependent barclamps is already queued, or has never run successfully.  These reasons should bubble up in the UI (web _and_ CLI / REST API), otherwise proposals can remain stuck in the queued state for no obvious reason.

This will require modifications to the following methods in `app/models/service_object.rb`:
- `elements_not_ready`
- `add_pending_elements`
- `queue_proposal`
- `process_queue`

I could probably fix this myself, but am somewhat hampered by lack of explanation in the code regarding what certain variables / data structures which get passed around are supposed to contain, in particular `delay` and `pre_cached_nodes`.  If someone could shed light on these, I'm more likely to be able to fix it.  Thanks!

**Comment by aspiers** [comment]
@bkutil @tboerger and another.  At least we have the deployment queue view now, but it should be visible from other pages too.

**Comment by bkutil** [comment]
Reasons for queuing are not part of the flash messages, aren't they? (e.g. waiting for nodes x, y to become ready).

**Comment by aspiers** [comment]
Yeah maybe the flash is good enough, but are all the cases are already covered?

**Comment by bkutil** [comment]
Good point. I don't think they are. Only 'node missing' is. 'Another role is in the queue', IMHO.

**Comment by tboerger** [comment]
A better flash message should be cool.
